character-controller.js

```diff
@@ -40,6 +40,7 @@ import {
   defaultPlayerBio,
 } from './ai/lore/lore-model.js';
 import {makeId, clone, unFrustumCull, enableShadows} from './util.js';
+import gameManager from './game.js';

```

Comment:
I don't think the character controller should ever contact the `game` manager.

There is something wrong with the way this code is structured.

Reply:
Added a new state `needResetUseTime` and modified checking fn of `actionInterpolants.use` . https://github.com/webaverse/app/pull/2804/commits/5c8c30ac3422242d4cb9481b6c05e28fbcd50769
Now can run `menuEndUse` and `menuStartUse` both before `updateInterpolation()`,
No need the order: `menuEndUse` -> `updateInterpolation()` -> `menuStartUse`.

So now can put them out of `updateAvatar()`.
Moved from `character-controller.js` to `game.js` directly, no need import `game` any more. https://github.com/webaverse/app/pull/2804/commits/f9f0982f7079bf395234a63632da5fc311cde8cd